ledger/appcow.go

```diff
@@ -467,7 +467,12 @@ func (cb *roundCowState) BuildEvalDelta(aidx basics.AppIndex, txn *transactions.
 				if err != nil {
 					return basics.EvalDelta{}, err
 				}
-				evalDelta.LocalDeltas[addrOffset] = sdelta.kvCow.serialize()
+
+				d := sdelta.kvCow.serialize()
+				// ignore empty deltas
+				if len(d) != 0 {
```

Comment:
while it's probably correct, I think that you ignored the case where `len(d) == 0` and `len(evalDelta.LocalDeltas[addrOffset]) != 0`.
i.e. you would need to add a `delete(evalDelta.LocalDeltas, addrOffset)` here. I'm not sure how this would work in the previous release, and it could be a  non-issue here as well.

Reply:
1. all txn have independent eval delta
2. there is only one kv per addr+appId combination
3. I'll add couple more tests to document this behavior